### PR TITLE
S4703527YOLOv8

### PR DESCRIPTION
**Pull request by mraula**
Author: Raul Movila
Student number: 47035276

Problem:  3. Detect lesions within the ISIC 2017/8 data set (see Appendix for link) with the or a detection network
such as the YOLOv7 or newer (pre-trained models are allowed) with all detections having a minimum
Intersection Over Union of 0.8 on the test set and a suitable accuracy for classification. [Normal Difficulty]

Training:
The original data had to be augmented to create the necessary box plots for the Yolov8 model. The Data was split in a 70/10/20 training, validation and test split. 

Fine-tuning the model is the most important part of the task. The model itself is not very complicated to load but most of the time will be taken retraining the model on different parameters to achieve our desired classification accuracy and IoU. After fine-tuning the best model was saved and the following loss values recorded:

Box Loss: 0.54229
Class Loss: 0.30157
Distribution Focal Loss (DFL): 0.97581

Validation:

Precision: 0.872 Precision measures how many predicted positive were actually correct.
Recall: 0.860 Recall measures the proportion of actual positives that were correctly identified by the model.

Final Test Results:

Metric | Value
------- | --------
Precision | 0.855
Recall | 0.873
IoU avg | 0.815

### CODE REVIEW

**Comment by shaivikaaaa** [comment]
**This is an initial inspection**

Difficulty : Normal
Task : 3

1) Recognition Problem:

- code structure is ok
- file structure is ok
- comments are not elaborative 
- no graph plots uploaded

2) Commit Log:

- Commit message are not as elaborative  
- Commits are not progressive and done in 1 day

3) Documentation:
README is good

4) Pull Request

- PR does not satisfy the requirement 

Feedback:

- PR can be better and more detailed then what you have now
- mention train/test % 
- add graph plots

**Comment by mraula** [comment]
Hello, 

I updated my PR to include more information as well as the train/test %. I am a bit confused where I should add the graph plots I have them in my figures folder, should I add any relevant plots to the PR?

Also my commits are done in two days as I first did the Siamese network but after 2 weeks of training I think I had some problems with my embedding and my validation was stuck at 0.7 for pair matching and test was at 0.5 with melanoma classification. The Yolo task took Two days of research on the model then a day of getting the database ready and loading up the Yolov8 model which were committed every time i achieved something new. Then I retained the model around 11 different times with different hyper-parameters and Yolo models like YoloV11 and committed the best model.

Thank You.

**Comment by aniketgupta17** [comment]
Observational Feedback

Pull Request:
Correctly created the Pull request from Topic Recognition Branch .
The pull request should include a clear description about the file structure .
Previous feedback incorporated .

File Organizing: Well-organized files.

Commit Log:
Commit messages are not  progressive for the Recognition Problem solved using 4 files .
Commits are not regularly made showing no logical development and completed within 2 days

Documentation:
Code comments and docstrings are included.
Proper GitHub markdown formatting is used, with organized headings, lists, and code blocks.
Should add about Conclusion and future improvements in the ReadMe.

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>Good solution to problem. </td><td></td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Normal. YOLO (Normal Difficulty)</td><td>-5</td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Some/Adequate Meaningful commit messages. Abstract.</td><td>-1</td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td>-2</td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Good. </td><td></td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF Checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). On time.</td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is good. </td><td></td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-9</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_
